corehq/ex-submodules/casexml/apps/case/sharedmodels.py

```diff
@@ -79,8 +79,26 @@ class CommCareCaseAttachment(LooselyEqualDocumentSchema, CaseAttachmentMixin, Un
     def content_type(self):
         return self.server_mime
 
+    @property
+    def is_present(self):
+        """
+        Helper method to see if this is a delete vs. update
+
+        NOTE this is related to but reversed logic from
+        `casexml.apps.case.xml.parser.CaseAttachment.is_delete`.
+        """
+        # Should this only use `attachment_src` since that is the
+        # only field checked by `from_case_index_update`?
+        return not (
+            self.identifier and
+            self.attachment_src is self.attachment_from is None
+        )
+
     @classmethod
     def from_case_index_update(cls, attachment):
+        # yet another is_deleted check? but different from the others;
+        # this one only looks at attachment_src instead of some combination
+        # of attachment_src, attachment_from, and attachment_name
         if attachment.attachment_src:
```

Comment:
Guessing this is OK in our implementation since we don't support `inline` attachments. Would be better to standardize though.